test/test_fake_tensor.py

```diff
@@ -554,6 +554,15 @@ def test_jagged_fake_to_fake_preserved(self):
         self.assertIs(t2.size(1).node.shape_env, t1.size(1).node.shape_env)
         self.assertEqual(str(t2.size(1)), str(t1.size(1)))
 
+    def test_fake_parent(self):
+        shape_env = ShapeEnv()
+        mode1 = FakeTensorMode(shape_env=shape_env)
+        mode2 = FakeTensorMode(parent=mode1, shape_env=shape_env)
```

Comment:
Should we just add an assert that if you initialize a `FakeTensorMode` with a parent, they must have the same ShapeEnv?

I suppose you could have a parent with a different ShapeEnv (or even without a ShapeEnv), but that seems like it adds a lot of degrees of freedom and I'm not sure what the benefit is.

Reply:
I can add the assert if it makes you happy. To me it does make sense to have non-matching shape env but it is true there is no concrete use case for it

Reply:
ok nvm then seems fine to leave